### DISCUSSION TITLE
when set c->is_closing = 1 in HTTP server, why WEB page is not crash

### DISCUSSION THREAD

**Discussion by yanruyue** [cesanta/mongoose#1910] [Ask questions, get answers] [answered]
I was trying to update mongoose from V6.17 to V7.8  in our HTTPs server.
When using V6.17, when c->flags |= MG_F_CLOSE_IMMEDIATELY is set, the Web page  will response with crash "This page isn’t working"
But in V7.8, when c->is_closing = 1 is set, the WEBUI seems waiting for a reply and will not response.
I'm very confused, as c->is_closing is set to be true, the connection should be closed immediately, so Web page should crash with "This page isn’t working", why the web page is still waiting for a reply?

Looking forward for your answer, Thank you very much.

**Comment by scaprile** [comment, marked as answer]
What web page ? What crash ? What does it do ? What does it expect ? Did you sniff your network ? Did you see HTTP/TCP traffic ?
If Mongoose is your server and you set the `is_closing` flag, the connection will be closed https://mongoose.ws/documentation/#connection-flags . If you need to send remaining data, set `is_draining` instead.
Check our tutorials here: https://mongoose.ws/tutorials/

**Reply by yanruyue** [reply]
The web page is the client to the HTTP server, and Mongoose is the HTTP server.
The point is, I set the is_closing flag, but the packages captured by wireshark is strange. 
No FIN package was found from server side. Instead, lots TCP zero window and TCP window full shows up.
172.16.60.89 side is the HTTP server(Mongoose)
172.16.60.80 side is the client(Web page)
![image](https://user-images.githubusercontent.com/84427169/206336584-f4aa9e45-eebc-400a-a06f-503fd785532b.png)


And I want web page to show
![image](https://user-images.githubusercontent.com/84427169/206333822-f80308ea-c541-442b-aaca-ab0e610c1840.png)

**Reply by scaprile** [reply]
I suggest you follow the documentation and examples. Are you properly calling the event manager ?
If you still can't make it work, then please open an issue with the minimum piece of code showing your problem. Follow the issue template providing all the required information. Please no pictures, attach a pcap file.

**Reply by yanruyue** [reply]
Thank you very much for your reply, It turns out it was me, I ignored mongoose.c defined closesocket(x) to close(x) in Line3979, so don't call my own closesocket() function and causes this problem.
Thanks again for your support!
